.github/workflows/python.yaml

```diff
@@ -1,28 +1,50 @@
 name: Python package
 
-on: [push]
+on:
+  workflow_dispatch:
+  pull_request:
+    branches: [feature/capgen, main]
 
 jobs:
   build:
-
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        python-version: [3.7]
+        python-version: ['3.7', '3.8', '3.9', '3.10', '3.11']
 
     steps:
-    - uses: actions/checkout@v2
+    - uses: actions/checkout@v3
     - name: Set up Python ${{ matrix.python-version }}
-      uses: actions/setup-python@v2
+      uses: actions/setup-python@v4
       with:
         python-version: ${{ matrix.python-version }}
     - name: Install dependencies
       run: |
         python -m pip install --upgrade pip
-        pip install flake8 pytest
-        if [ -f requirements.txt ]; then pip install -r requirements.txt; fi
+        pip install pytest
     - name: Test with pytest
-      if: github.repository == 'NCAR/ccpp-framework' # Only run on main repo
       run: |
         export PYTHONPATH=$(pwd)/scripts:$(pwd)/scripts/parse_tools
-        pytest
+        pytest -v
+
+  doctest:
+    if: github.event_name == 'pull_request' || github.event_name == 'workflow_dispatch' || github.repository == 'NCAR/ccpp-framework'
```

Comment:
Do we still need the complicated logic here?

Reply:
good catch! removed.